rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java

```diff
@@ -558,6 +571,57 @@ public void setStatus(NodeSourceStatus status) {
         this.infrastructureManager.setPersistedNodeSourceData(NodeSourceData.fromNodeSourceDescriptor(this.descriptor));
     }
 
+    public LinkedHashMap<String, String> getAdditionalInformation() {
+        return additionalInformation;
+    }
+
+    public void putAndPersistAdditionalInformation(String key, String value) {
+        String valueToUpdate = this.additionalInformation.get(key);
+        if (valueToUpdate == null || (valueToUpdate != null && !valueToUpdate.equals(value))) {
+
+            // Put additional information and make it visible from the rm portal by triggering an event
+            this.additionalInformation.put(key, value);
+            this.descriptor.getAdditionalInformation().put(key, value);
+            this.monitoring.nodeSourceEvent(new RMNodeSourceEvent(RMEventType.NODESOURCE_UPDATED,
+                                                                  this.administrator.getName(),
+                                                                  this.name,
+                                                                  this.getDescription(),
+                                                                  this.additionalInformation,
+                                                                  this.administrator.getName(),
+                                                                  this.getStatus().toString()));
+            // Persist additional information
+            persistAdditionalInformation();
```

Comment:
persist should be called before sending the event

Reply:
Ok